test/cusolver/dense.jl

```diff
@@ -421,7 +421,9 @@ k = 1
         @test Array(h_q) ≈ Array(q)
         @test collect(CuArray(h_q)) ≈ Array(q)
         @test Array(h_r) ≈ Array(r)
-        @test CuArray(h_q) ≈ convert(typeof(d_A), h_q)
+        if VERSION >= v"1.8-"
+            @test CuArray(h_q) ≈ convert(typeof(d_A), h_q)
```

Comment:
Why wouldn't we want this to work on <1.8?

Reply:
I was about to ask a question which is strongly linked to this: why are most of the functions defined in [lib/cusolver/linalg.jl](https://github.com/JuliaGPU/CUDA.jl/blob/e0710be4bf422e2468f63a4712ed4376f2717e4d/lib/cusolver/linalg.jl) only available for Julia >= 1.8? Most of the QR, SVD and LU related functions are gated behind tests and are only available on Julia >= 1.8.

Reply:
Because https://github.com/JuliaLang/julia/pull/42594 is only part of 1.8; before, we used to have custom structures for factorizations, which were only implemented for a couple of them (e.g. CuQR)

Reply:
Ok, thanks for the link! It's clearer now, I was confused about the different implementations for versions under 1.8 and versions above 1.8.
I tried to fix it for versions under 1.8, and ran into another bug. Running 
```
A = CUDA.rand(10,4)
q,_ = qr(A)
Matrix{Float32}(q)
```
seems to fail for versions under 1.8, as the [default fallback](https://github.com/JuliaLang/julia/blob/a41ae5bb911819f1b3011a654e2bb68c39be8115/stdlib/LinearAlgebra/src/qr.jl#L575) is apparently called. I think I've managed to fix it, but I may be doing one too many conversion.